frontend/middleware_validatesubscription.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/ARO-HCP/internal/api/subscription"
+)
+
+const (
+	UnregisteredSubscriptionStateMessage = "The subscription %s is not registered for this provider %s. Please re-register the subscription."
```

Comment:
are these messages the same as ARO Classic?

Reply:
My line of thinking: 
- We should consolidate these somewhere at some point, but not now.  
- BU should be good with error messages.  If they're the same, this is good.

Reply:
They are not the same. I can take a look at classic and align with them for now.

Reply:
would a `//todo align with aro classic` be a good way to capture these items temporarily for future consolidation?  Or not worth the effort?

Reply:
might be best to do this type of consolidation when we have more errors to align, for now I updated the ones that do exist in both to match classic.